File: quicksort.py
def quicksort(arr:list):
    # print(arr)
    if len(arr) < 2:
        # print(arr)
        return arr
    else:
        pivot = arr[0]
        left = [i for i in arr[1:] if i <= pivot]
        right = [i for i in arr[1:] if i > pivot]
        print(f'{left} <=> [{pivot}] <=> {right}')
        return quicksort(left) + [pivot] + quicksort(right)
    
arr = [3,1,2,4,5,8,7,6,9,10]
print(quicksort(arr))

File: quicksort.c
#include <stdio.h>

int *qsort(int *arr, int arrsize);

int *qsort(int *arr, int arrsize){
    if (arrsize < 2){
        return arr;
    } else{
        int mid = arrsize/2;
        int pivot = arr[mid];
        int left[mid], right[mid];
        for (int i = 0; i < mid; i++)
        {
            left[i] = arr[i];
            right[i] = arr[i + mid];
        }
        int rsize = (arrsize % 2 == 0) ? (mid - 1): mid;
        return qsort(left, mid);
    }
}